malariagen_data/ag3.py

```diff
@@ -713,61 +726,113 @@ def snp_effects(self, transcript, site_mask):
-        stop = feature[4]
-        strand = feature[6]
+        """
+        # get feature details (don't need to set up an annotator here)
+        geneset = self.geneset()
+        geneset = geneset.set_index('ID')
+        feature = geneset.loc[transcript]
+        contig = feature.seqid
+        start = feature.start
+        stop = feature.end
+        strand = feature.strand
 
         print(
             f"transcript : {transcript}\nchromosome : {contig} \nstart : {start}\nstop : {stop}"
             f"\nstrand : {strand}"
         )
 
         # grab pos, ref and alt for chrom arm from snp_sites
-        sites = self.snp_sites(contig=contig, site_mask=site_mask)
+        pos, ref, alt = self.snp_sites(contig=contig, site_mask=site_mask)
 
         # sites are dask arrays, turn pos into sorted index
-        pos = allel.SortedIndex(sites[0].compute())
+        pos = allel.SortedIndex(pos.compute())
         # locate transcript range
         loc = pos.locate_range(start, stop)
 
         # we want to grab all metadata then get idx for samples we want
-        # what granularity do we want here - country+site+year TODO
         df_meta = self.sample_metadata(
-            sample_sets="v3_wild", species_calls=("20200422", "aim")
+            sample_sets=sample_sets, species_calls=species_calls
         )
 
-        # get genotypes - chop to loc, chop to pop_idx TODO
+        # get genotypes
         gt = self.snp_genotypes(
             contig=contig,
-            sample_sets="v3_wild",
+            sample_sets=sample_sets,
             field="GT",
             site_mask=site_mask,
-            site_filters="dt_20200416",
+            site_filters=site_filters,
         )
-        gt = gt[loc].compute()
-
-        # count alleles - should we calculate and add these to gcs like previous phases?
-
-        # counts to frequencies
 
-        # build and return dataframe
+        # chop everything to loc
+        gt = gt[loc].compute()
+        pos = pos[loc]
+        ref = ref[loc].compute()
+        alt = alt[loc].compute()
+
+        # count alleles
+        afs = dict()
+        for pop, query in populations.items():
+            loc_pop = df_meta.eval(query).values
+            gt_pop = da.compress(loc_pop, gt, axis=1)
+            ac_pop = allel.GenotypeDaskArray(gt_pop).count_alleles(max_allele=3).compute()
+            afs[pop] = ac_pop.to_frequencies()
+
+        # set up columns
+        cols = {
+            'position': np.repeat(pos, 3),
+            'ref_allele': np.repeat(ref.astype('U1'), 3),
+            'alt_allele': alt.astype('U1').flatten(),
+        }
+
+        for pop in populations:
+            cols[pop] = afs[pop][:, 1:].flatten()
+
+        # build df
+        df = pandas.DataFrame(cols)
+
+        # drop invariants
+        if drop_invariants:
+            df['dropping'] = df[populations].sum(axis=1)
+            df = df[df.dropping > 0]
+            df.drop('dropping', axis=1, inplace=True)
```

Comment:
```suggestion
            loc_variant = df[populations].sum(axis=1) > 0
            df = df[loc_variant]
```

Simplify, there's no need to add then remove a column here.